lib/nexpose/device.rb

```diff
@@ -208,4 +223,50 @@ def self.parse_json(json)
   #
   class IncompleteAsset < CompletedAsset
   end
+
+
+  # Summary object of a scan for a particular asset.
+  #
+  class AssetScan
+    # Unique identifier of an asset.
+    attr_reader :asset_id
+    # IP address of the asset.
+    attr_reader :ip
+    # Host name of the asset, if discovered.
+    attr_reader :host_name
+    # Site name where the scan originated.
+    attr_reader :site_name
+    # Unique identifier for the site where the scan originated.
+    attr_reader :site_id
+    # Unique identifier for the scan.
+    attr_reader :scan_id
+    # Time when the asset finished scanning.
+    attr_reader :end_time
+    # Number of vulnerabilities discovered on the asset.
+    attr_reader :vulns
+    # Operating system fingerprint of the asset.
+    attr_reader :os
+    # Name of the scan engine used for the scan.
+    attr_reader :engine_name
+
+    # Internal constructor to be called by #parse_json.
+    def initialize(&block)
+      instance_eval(&block) if block_given?
+    end
+
+    def self.parse_json(json)
+      new do
+        @asset_id = json['assetID'].to_i
+        @scan_id = json['scanID'].to_i
+        @site_id = json['siteID'].to_i
+        @ip = json['ipAddress']
+        @host_name = json['hostname']
+        @os = json['operatingSystem']
+        @vulns = json['vulnCount']
+        @end_time = Time.at(json['completed'].to_i / 1000)
```

Comment:
Do not use `Time.at` without zone. Use one of `Time.zone.at`, `Time.at.current`, `Time.at.in_time_zone`, `Time.at.utc`, `Time.at.getlocal`, `Time.at.iso8601`, `Time.at.jisx0301`, `Time.at.rfc3339`, `Time.at.to_i`, `Time.at.to_f` instead.